src/main/java/org/apache/sysml/runtime/controlprogram/paramserv/LocalPSWorker.java

```diff
@@ -31,67 +32,70 @@
 	protected static final Log LOG = LogFactory.getLog(LocalPSWorker.class.getName());
 
 	public LocalPSWorker(long workerID, String updFunc, Statement.PSFrequency freq, int epochs, long batchSize,
-			ListObject hyperParams, ExecutionContext ec, ParamServer ps) {
-		super(workerID, updFunc, freq, epochs, batchSize, hyperParams, ec, ps);
+			ExecutionContext ec, ParamServer ps) {
+		super(workerID, updFunc, freq, epochs, batchSize, ec, ps);
 	}
 
 	@Override
 	public void run() {
+		try {
+			long dataSize = _features.getNumRows();
 
-		long dataSize = _features.getNumRows();
+			for (int i = 0; i < _epochs; i++) {
+				int totalIter = (int) Math.ceil(dataSize / _batchSize);
+				for (int j = 0; j < totalIter; j++) {
+					// Pull the global parameters from ps
+					// Need to copy the global parameter
+					ListObject globalParams = ParamservUtils.copyList((ListObject) _ps.pull(_workerID));
```

Comment:
Well, I think the problem is that agg service will broadcast the references of the model to all the workers. And then when the worker finish its iteration, it will destruct this pulled model which is also referenced by other workers. Hence, to avoid impacting the global model in other workers, I'd like to copy the global model in each worker so that it can have its own local model.

Reply:
Thanks for the clarification - the function call and list indexing should already take care of disabling the cleanup. If this is not working then it simply needs fixing. I'll have a look during the merge.